File: LSP+ISP/C#_Final/MarsRoverKata/Commands/TurnRightCommand.cs
namespace MarsRoverKata.Commands
{
    public class TurnRightCommand : ICommand
    {
        private IMovable Movable { get; set; }

        public TurnRightCommand(IMovable movable)
        {
            Movable = movable;
        }

        public bool Execute()
        {
            return Movable.TurnRight();
        }
    }
}

File: DIP/Java_Final/MarsRoverKata/src/main/java/MarsRoverKata/Commands/TurnRightCommand.java
package MarsRoverKata.Commands;

import MarsRoverKata.IMovable;

public class TurnRightCommand implements ICommand {
    private IMovable movable;

    public IMovable getMovable() {
        return movable;
    }

    public void setRover(IMovable movable) {
        this.movable = movable;
    }

    public TurnRightCommand(IMovable movable) {
        this.movable = movable;
    }

    public boolean execute() {
        return movable.turnRight();
    }

}
